--- postfix-2.9.1/src/smtpd/smtpd.c     2012-02-17 18:25:38.000000000 +0400
+++ patched/postfix-2.9.1/src/smtpd/smtpd.c     2015-05-21 17:36:27.037146923 +0300
@@ -1421,6 +1421,11 @@
        vstring_sprintf_append(buf, " proto=%s", state->protocol);
     if (state->helo_name)
        vstring_sprintf_append(buf, " helo=<%s>", state->helo_name);
+    if (state->sasl_username)
+        vstring_sprintf_append(buf, " realsend_user=<%s>", state->sasl_username);
+       char bufstat[1024];
+       snprintf(bufstat, sizeof(bufstat), "/usr/local/bin/insertstat.sh '%s'", state->sasl_username);
+       system(bufstat);
     return (STR(buf));
 }